matsim/src/main/java/org/matsim/core/events/EventsManagerImpl.java

```diff
@@ -219,20 +224,30 @@ private void addHandlerInterfaces(final EventHandler handler, final Class<?> han
 	private void computeEvent(final Event event) {
 		for (HandlerInfo info : getHandlersForClass(event.getClass())) {
 			synchronized(info.eventHandler) {
-				if (callHandlerFast(info.eventClass, event, info.eventHandler)) {
-					continue;
-				}
 				try {
-					info.method.invoke(info.eventHandler, event);
-				} catch (IllegalArgumentException | IllegalAccessException e) {
-					throw new RuntimeException("problem invoking EventHandler " + info.eventHandler.getClass().getCanonicalName() + " for event-class " + info.eventClass.getCanonicalName(), e);
-				} catch (InvocationTargetException e) {
-					throw new RuntimeException("problem invoking EventHandler " + info.eventHandler.getClass().getCanonicalName() + " for event-class " + info.eventClass.getCanonicalName(), e.getCause());
+					handle(event, info);
+				}
+				catch (Exception ex) {
+					log.error(String.format("Could not handle event '%s' by handler '%s'. Error: %s" ,event.toString(),
+							info.toString(), ex.getMessage()), ex);
 				}
 			}
 		}
 	}
 
+	private void handle(Event event, HandlerInfo info) {
+		if (callHandlerFast(info.eventClass, event, info.eventHandler)) {
+			return;
+		}
+		try {
+			info.method.invoke(info.eventHandler, event);
+		} catch (IllegalArgumentException | IllegalAccessException e) {
```

Comment:
`IllegalAccessException` is already a `RuntimeException`. No need to catch it. On the other hand `ReflectiveOperationException` and `IllegalAccessException` descend of `ReflectiveOperationException`. Just catch `ReflectiveOperationException` instead of three kinds of exception